aten/src/ATen/cpu/vec256/vec256_float_neon.h

```diff
@@ -642,6 +642,853 @@ Vec256<float> inline fmadd(const Vec256<float>& a, const Vec256<float>& b, const
+  operator float32x4x2_t() const {
+    return values;
+  }
+  template <int64_t mask>
+  static Vec256<float> blend(const Vec256<float>& a, const Vec256<float>& b) {
+    // 0 initialize vec since it will also serve as mask that
+    // will be filled first and then we will apply bit select op.
+    Vec256<float> vec(0);
+    // 0.
+    // Construct mask first.
+    vec.values.val[0] = fill_mask<0, (mask & 0x01)!=0>(vec.values.val[0]);
+    vec.values.val[0] = fill_mask<1, (mask & 0x02)!=0>(vec.values.val[0]);
+    vec.values.val[0] = fill_mask<2, (mask & 0x04)!=0>(vec.values.val[0]);
+    vec.values.val[0] = fill_mask<3, (mask & 0x08)!=0>(vec.values.val[0]);
+    __asm__ (
+        "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+        : [res_mask] "+w" (vec.values.val[0])
+        : [in_b] "w" (b.values.val[0]), [in_a] "w" (a.values.val[0])
+        :);
+    // 1.
+    // Construct mask first.
+    vec.values.val[1] = fill_mask<0, (mask & 0x10)!=0>(vec.values.val[1]);
+    vec.values.val[1] = fill_mask<1, (mask & 0x20)!=0>(vec.values.val[1]);
+    vec.values.val[1] = fill_mask<2, (mask & 0x40)!=0>(vec.values.val[1]);
+    vec.values.val[1] = fill_mask<3, (mask & 0x80)!=0>(vec.values.val[1]);
+    __asm__ (
+        "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+        : [res_mask] "+w" (vec.values.val[1])
+        : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+        :);
+    return vec;
+  }
+  static Vec256<float> blendv(const Vec256<float>& a, const Vec256<float>& b,
+                              const Vec256<float>& mask) {
+    // TODO
+    // NB: This requires that each value, i.e., each uint value,
+    // of the mask either all be zeros or all be 1s.
+    // We perhaps need some kind of an assert?
+    // But that will affect performance.
+    Vec256<float> vec(mask.values);
+    __asm__ (
+        "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+        : [res_mask] "+w" (vec.values.val[0])
+        : [in_b] "w" (b.values.val[0]), [in_a] "w" (a.values.val[0])
+        :);
+    __asm__ (
+        "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+        : [res_mask] "+w" (vec.values.val[1])
+        : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+        :);
+    return vec;
+  }
+  template<typename step_t>
+  static Vec256<float> arange(float base = 0.f, step_t step = static_cast<step_t>(1)) {
+    const Vec256<float> base_vec(base);
+    const Vec256<float> step_vec(step);
+    const Vec256<float> step_sizes(0, 1, 2, 3, 4, 5, 6, 7);
+    return fmadd(step_sizes, step_vec, base_vec);
+  }
+  static Vec256<float> set(const Vec256<float>& a, const Vec256<float>& b,
+                           int64_t count = size()) {
+    switch (count) {
+      case 0:
+        return a;
+      case 1:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_low = {0xFFFFFFFF, 0x0, 0x0, 0x0};
+          vec.values.val[0] = (float32x4_t)mask_low;
```

Comment:
It seems to me what this does is take UINT_MAX (i.e. 0xFFFFFFFF == 4294967295) and store the equivalent IEEE-754 FP32 bit pattern that's closest to representing 4294967295.0f (i.e. 0x4F800000) into val[0].  Is that what you intended?  If you wanted the 0xFFFFFFFF bit pattern you should either use `vreinterpretq_f32_u32` or `*(const float32x4_t*)(&mask_low)`.

Reply:
Since the type is `uint32x4_t` I am not sure how meaningful it is to cast to a different pointer type. On the other hand use of intrinsics was resulting in seg fault. I just look the intrinsic up in arm_neon.h and it was doing just `(float32x4_t)`, so I replaced that with it. Although not sure why the similar use of intrinsic elsewhere did not cause problem.

Reply:
Interesting.